src/logging.cc

```diff
@@ -1416,8 +1416,34 @@ bool LogCleaner::IsLogFromCurrentProject(
     }
   }
 
+  // cleaned_base_file contains .<severity>. at the end
+  // If we check it against filepath it will only allow files
+  // with that severity. As we want to delete old logs despite their severity,
+  // we need to replace <severity> in cleaned_base_file and find matching one. 
+
+  // Erase severity name from cleaned_base_filename
+  // Delete last dot to be able to use `find_last_of` function
+  cleaned_base_filename.pop_back();
+  int dot_pos = cleaned_base_filename.find_last_of('.');
+  cleaned_base_filename.erase(dot_pos + 1, std::string::npos);
+
+  // Loop through severity names and check whether
+  // filepath contains cleaned_base_filename_with_severity
+  bool found_file = false;
+  for (const auto& severity : LogSeverityNames) {
+    string cleaned_base_filename_with_severity = cleaned_base_filename + severity;
```

Comment:
```suggestion
```suggestion
  const string base_filename = cleaned_base_filename.substring(0, dot_pos);
  bool found_file = false;
  for (const auto& severity : LogSeverityNames) {
    string cleaned_base_filename_with_severity = base_filename + severity;
```